valhalla/baldr/location.h

```diff
@@ -27,6 +28,26 @@ struct Location {
 
   enum class PreferredSide : uint8_t { EITHER, SAME, OPPOSITE };
 
+  /**
+   * Optional filters supplied in the request.
+   */
+  struct SearchFilter {
```

Comment:
This struct is a mirror of the protobuf message, per @kevinkreiser there's a larger refactor to be done here for Loki to use the proto generated valhalla::Location directly. I didn't tackle that refactor in this PR.

Reply:
Could we add a HUGE TODO and CAUTION to the docstring with what your comment just said?